### PR TITLE
Implement proper macOS version check

### PR DESCRIPTION
**Pull request by nastys**

### CODE REVIEW

File: rpcs3/util/sysinfo.cpp [hunk 1]
```diff
@@ -11,9 +11,12 @@
 #include "stringapiset.h"
 #else
 #include <unistd.h>
+#include <sys/resource.h>
```
Comment (Megamouse) [inline rpcs3/util/sysinfo.cpp:14]: Why is this include added for all other compilers?

File: rpcs3/Emu/CMakeLists.txt [hunk 2]
```diff
@@ -59,6 +59,11 @@ target_sources(rpcs3_emu PRIVATE
     ../../Utilities/Thread.cpp
     ../../Utilities/version.cpp
 )
+if(APPLE)
+	target_sources(rpcs3_emu PRIVATE
+	    ../util/sysinfo_darwin.mm
+	)
+endif()
```
Comment (Megamouse) [inline rpcs3/Emu/CMakeLists.txt:66]:
```suggestion
if(APPLE)
    target_sources(rpcs3_emu PRIVATE
        ../util/sysinfo_darwin.mm
    )
endif()
```

File: rpcs3/util/sysinfo.cpp [hunk 3]
```diff
@@ -342,6 +354,13 @@ std::string utils::get_OS_version()
 	fmt::append(output,
 		"Operating system: Windows, Major: %lu, Minor: %lu, Build: %u, Service Pack: %s, Compatibility mode: %llu",
 		version_major, version_minor, build, has_sp ? holder.data() : "none", compatibility_mode);
+#elif defined (__APPLE__)
+	const int majorVersion = Darwin_Version::getNSmajorVersion();
+	const int minorVersion = Darwin_Version::getNSminorVersion();
+	const int patchVersion = Darwin_Version::getNSpatchVersion();
+
+	fmt::append(output, "Operating system: macOS, Version: %i.%i.%i",
```
Comment (Megamouse) [inline rpcs3/util/sysinfo.cpp:362]: afaict, %d and %i do the same.
So I'd use %d for overall consistency

File: rpcs3/util/sysinfo.cpp [hunk 4]
```diff
@@ -342,6 +354,13 @@ std::string utils::get_OS_version()
 	fmt::append(output,
 		"Operating system: Windows, Major: %lu, Minor: %lu, Build: %u, Service Pack: %s, Compatibility mode: %llu",
 		version_major, version_minor, build, has_sp ? holder.data() : "none", compatibility_mode);
+#elif defined (__APPLE__)
+	const int majorVersion = Darwin_Version::getNSmajorVersion();
+	const int minorVersion = Darwin_Version::getNSminorVersion();
+	const int patchVersion = Darwin_Version::getNSpatchVersion();
```
Comment (Megamouse) [inline rpcs3/util/sysinfo.cpp:360]: we generally use snake_case. (at least for new code)
So majorVersion should be major_version etc.

File: rpcs3/util/sysinfo_darwin.mm [hunk 5]
```diff
@@ -0,0 +1,21 @@
+#import <Foundation/Foundation.h>
```
Comment (kd-11) [inline rpcs3/util/sysinfo_darwin.mm:1]: This belongs in a different directory. iirc xcode stuff is objective-c/c++ right?

Comment (nastys) [reply rpcs3/util/sysinfo_darwin.mm:1]: Yes. This file is Objective-C++ (mixed Objective-C + C++).

**Pull request merged**